public/app/features/plugins/extensions/utils.tsx

```diff
@@ -159,3 +159,22 @@ function isRecord(value: unknown): value is Record<string | number | symbol, unk
 export function isReadOnlyProxy(value: unknown): boolean {
   return isRecord(value) && value[_isProxy] === true;
 }
+
+export function createExtensionLinkConfig<T extends object>(
+  config: Omit<PluginExtensionLinkConfig<T>, 'type'>
+): PluginExtensionLinkConfig {
+  const linkConfig = {
+    type: PluginExtensionTypes.link,
+    ...config,
+  };
+  assertIsLinkConfig(linkConfig);
+  return linkConfig;
+}
+
+function assertIsLinkConfig<T extends object>(
+  config: PluginExtensionLinkConfig<T>
+): asserts config is PluginExtensionLinkConfig {
+  if (config.type !== PluginExtensionTypes.link) {
+    throw Error('value is not a string');
```

Comment:
I'm not sure I'm understanding the usage of this function. It's not exported and only called above where the `linkConfig.type` passed in will always be `PluginExtensionTypes.link`. 🤔

Reply:
basically a quick hack just to stop betterer from complaining about doing `x as y`.

Reply:
Wanted to push something that I could get early feedback on.